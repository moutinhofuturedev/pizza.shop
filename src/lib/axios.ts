import axios, { AxiosError, AxiosInstance, AxiosResponse } from 'axios'

import { env } from '@/env'

export const api: AxiosInstance = axios.create({
  baseURL: env.VITE_BASE_URL,
  withCredentials: true,
})

// Intercepta requisições
api.interceptors.request.use(
  async (config) => {
    if (env.VITE_ENABLE_DELAY_API) {
      await new Promise((resolve) => setTimeout(resolve, 1000))
    }

    console.log('Solicitação enviada:', config.url)
    return config
  },
  (error) => {
    console.error('Erro na configuração da solicitação:', error)
    return Promise.reject(error)
  },
)

// Intercepta respostas
api.interceptors.response.use(
  (response: AxiosResponse): AxiosResponse => {
    console.log('Resposta de sucesso:', response.status, response.statusText)
    return response
  },
  (error) => {
    if (axios.isAxiosError(error)) {
      const axiosError = error as AxiosError

      if (axiosError.response) {
        // Erro na resposta da API
        console.error(
          'Erro na resposta da API:',
          axiosError.response.status,
          axiosError.response.statusText,
          axiosError.response.data,
        )
      } else if (axiosError.request) {
        // Erro na solicitação da API
        console.error('Erro na solicitação da API:', axiosError.request)
      } else {
        // Erro genérico ou erro ao configurar solicitação da API
        console.error(
          'Erro na configuração da solicitação:',
          axiosError.message,
          error,
        )
      }
    } else {
      // Erro que não é do Axios
      console.error('Erro desconhecido:', error)
    }
    return Promise.reject(error)
  },
)
