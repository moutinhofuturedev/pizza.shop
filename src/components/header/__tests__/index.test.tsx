import { QueryClientProvider } from '@tanstack/react-query'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import { queryClient } from '@/lib/react-query'

import { Header } from '..'

describe('<Header />', () => {
  it('it should render correctly', () => {
    const { container } = render(<Header />, {
      wrapper: ({ children }) => {
        return (
          <QueryClientProvider client={queryClient}>
            <MemoryRouter initialEntries={['/']}>{children}</MemoryRouter>
          </QueryClientProvider>
        )
      },
    })

    expect(container).toMatchSnapshot()
  })

  it('should display text "Inicio" and "Pedidos" in the header', () => {
    const wrapper = render(<Header />, {
      wrapper: ({ children }) => {
        return (
          <QueryClientProvider client={queryClient}>
            <MemoryRouter initialEntries={['/']}>{children}</MemoryRouter>
          </QueryClientProvider>
        )
      },
    })

    const homeLink = wrapper.getByText('Inicio')
    const orderLink = wrapper.getByText('Pedidos')

    expect(homeLink).toBeInTheDocument()
    expect(orderLink).toBeInTheDocument()
  })
})
