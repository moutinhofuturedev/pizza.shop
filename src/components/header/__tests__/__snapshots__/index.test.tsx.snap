// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > it should render correctly 1`] = `
<div>
  <div
    class="border-b"
  >
    <div
      class="flex h-16 items-center gap-6 px-6 max-md:justify-center max-md:gap-3"
    >
      <svg
        class="lucide lucide-pizza h-6 w-6 max-sm:hidden"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 11h.01"
        />
        <path
          d="M11 15h.01"
        />
        <path
          d="M16 16h.01"
        />
        <path
          d="m2 16 20 6-6-20A20 20 0 0 0 2 16"
        />
        <path
          d="M5.71 17.11a17.04 17.04 0 0 1 11.4-11.4"
        />
      </svg>
      <div
        class="shrink-0 bg-border w-[1px] h-6 max-sm:hidden"
        data-orientation="vertical"
        role="none"
      />
      <nav
        class="flex items-center space-x-4 lg:space-x-6"
      >
        <a
          class="flex items-center gap-2 text-sm font-medium text-muted-foreground hover:text-foreground data-[active=true]:text-foreground data-[active=true]:underline"
          data-active="true"
          href="/"
        >
          <svg
            class="lucide lucide-house h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
            />
            <path
              d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
            />
          </svg>
          Inicio
        </a>
      </nav>
      <nav
        class="flex items-center space-x-4 lg:space-x-6"
      >
        <a
          class="flex items-center gap-2 text-sm font-medium text-muted-foreground hover:text-foreground data-[active=true]:text-foreground data-[active=true]:underline"
          data-active="false"
          href="/orders"
        >
          <svg
            class="lucide lucide-utensils-crossed h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m16 2-2.3 2.3a3 3 0 0 0 0 4.2l1.8 1.8a3 3 0 0 0 4.2 0L22 8"
            />
            <path
              d="M15 15 3.3 3.3a4.2 4.2 0 0 0 0 6l7.3 7.3c.7.7 2 .7 2.8 0L15 15Zm0 0 7 7"
            />
            <path
              d="m2.1 21.8 6.4-6.3"
            />
            <path
              d="m19 5-7 7"
            />
          </svg>
          Pedidos
        </a>
      </nav>
      <div
        class="ml-auto flex items-center gap-2"
      >
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 w-10"
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          <svg
            class="lucide lucide-sun h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <path
              d="M12 2v2"
            />
            <path
              d="M12 20v2"
            />
            <path
              d="m4.93 4.93 1.41 1.41"
            />
            <path
              d="m17.66 17.66 1.41 1.41"
            />
            <path
              d="M2 12h2"
            />
            <path
              d="M20 12h2"
            />
            <path
              d="m6.34 17.66-1.41 1.41"
            />
            <path
              d="m19.07 4.93-1.41 1.41"
            />
          </svg>
          <svg
            class="lucide lucide-moon absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
            />
          </svg>
          <span
            class="sr-only"
          >
            Toggle theme
          </span>
        </button>
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 flex select-none items-center gap-2"
          data-state="closed"
          id="radix-:r5:"
          type="button"
        >
          <span
            class="font-medium max-sm:hidden"
          >
            <div
              class="animate-pulse rounded-md bg-muted h-4 w-40"
            />
          </span>
          <svg
            class="lucide lucide-chevron-down h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
